### PR TITLE
Clear target dir

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
cc @pietroalbini 

r? @QuietMisdreavus

### CODE REVIEW

File: src/docbuilder/chroot_builder.rs [hunk 1]
```diff
@@ -237,6 +237,19 @@ impl DocBuilder {
                 .join("doc");
             let _ = remove_dir_all(crate_doc_path);
         }
+
+        if let Ok(fs) = System::new().mount_at(".") {
+            let disk_free = 100.0 - (fs.free.as_usize() as f32 / fs.total.as_usize() as f32);
+            if disk_free < 20.0 {
```
Comment (pietroalbini) [inline src/docbuilder/chroot_builder.rs]: Maybe extract this into a constant?

File: src/docbuilder/chroot_builder.rs [hunk 2]
```diff
@@ -237,6 +237,19 @@ impl DocBuilder {
                 .join("doc");
             let _ = remove_dir_all(crate_doc_path);
         }
+
+        if let Ok(fs) = System::new().mount_at(".") {
```
Comment (pietroalbini) [inline src/docbuilder/chroot_builder.rs]: I remember having trouble with this on Crater when `.` wasn't a mount point and I had [to do this](https://github.com/rust-lang/crater/blob/c0587168029236cce79847ad0819b76049004963/src/runner/worker.rs#L203-L220). Can you double check everything works?

**Comment by Mark-Simulacrum** [comment]
Review nits fixed and verified that this works locally (by setting allowed disk usage to 20% or so)

**Comment by onur** [comment]
thanks @Mark-Simulacrum but I think 20% might not be enough. Some crates (stm32) generate more than 10 GB documentation (no idea about their build artifacts) and since our disk space is really limited now, maybe we can increase it to at least 30 or 40 percent.

**Comment by Mark-Simulacrum** [comment]
We're currently sitting at 50 GB free on the server (out of 100 GB total). The way the logic is implemented, all 50 GB can be used by a single crate compilation (AFAICT), but before/after each crate build we will clear out the target directory if the usage exceeds 80GB. So, basically, this isn't making things any worse for an individual crate compilation.

**Comment by onur** [comment]
Then its great :+1:

**Comment by onur** [comment]
I am merging this before we ran out of space again :joy:

**Pull request merged**